work16/index.js

```diff
@@ -0,0 +1,248 @@
+                    const liElement = liElements[categoryNewsArticleDataIndex];
+
+                    liElement.append(createCommentIcon(commentArray));
+                }
+            })
+        })
+    })
+}
+
+const removeCommentIcon = () => {
+    const commentIconWrapElements = document.querySelectorAll(".commentIcon_wrap");
+    for (let k = 0; k < commentIconWrapElements.length; k++) {
+        commentIconWrapElements[k].remove();
+    }
+}
+
+const changeCategoryTitle = (newsArticleData, categoryNewsArticleDataIndex) => {
+    const currentNewsContentAncorElements = document.querySelectorAll(`section[aria-hidden="false"] a`);
+    currentNewsContentAncorElements[categoryNewsArticleDataIndex].textContent = newsArticleData.title;
+}
+
+const changeCategoryImg = (newsCategoryImg) => {
+    const currentNewsContentImgElem = document.querySelector(".tabTopicImg > img");
+    currentNewsContentImgElem.src = newsCategoryImg;
+}
+
+const changeCategoryConetent = (newsCategoryIndex, clickedTabElement) => {
+    const tabTopicIdName = clickedTabElement.id;
+    const currentNewsSectionElem = document.querySelector(`section[aria-hidden="false"]`);
+
+    currentNewsSectionElem.id = `${"tabpanelTopics" + (newsCategoryIndex + 1)}`;
+    currentNewsSectionElem.setAttribute("aria-labelledby", tabTopicIdName);
+}
+
+const renderArticle = (newsContentsData) => {
+    const ul = document.createElement("ul");
+    const fragment = document.createDocumentFragment();
+
+    newsContentsData.forEach(newsArticleData => {
+        const li = document.createElement("li");
+        const h1 = document.createElement("h1");
+        const a = document.createElement("a");
+
+        const title = newsArticleData.title;
+        const commentArray = newsArticleData.comments;
+
+        a.href = "#";
+        a.textContent = title;
+
+        h1.appendChild(a)
+        li.appendChild(h1);
+
+        if (commentArray.length !== 0) {
+            li.append(createCommentIcon(commentArray));
+        }
+        fragment.appendChild(li);
+    })
+    ul.appendChild(fragment);
+
+    return ul;
+}
+
+const createCommentIcon = (commentArray) => {
+    const div = document.createElement("div");
+
+    div.classList = "commentIcon_wrap";
+    div.insertAdjacentHTML("beforeend",
+        `<span class="fa-layers fa-fw">
+            <i class="fa fa-light fa-comment faa-vertical animated"></i>
+            <span class= "fa-layers-counter" > ${commentArray.length}</span>
+        </span>`
+    );
+
+    return div;
+}
+
+const createSection = (tabTopics, index) => {
+    const section = document.createElement("section");
+    const tabTopicElem = document.getElementById(tabTopics[index].id);
+    const tabIdName = tabTopicElem.id;
+
+    section.id = `${"tabpanelTopics" + (index + 1)} `;
+    section.setAttribute("aria-labelledby", tabIdName);
+    section.setAttribute("aria-hidden", "false");
```

Comment:
Q.Is `aria-hidden` necessary?
Looking at the code, I feel that the content portion is always visible and based on the rendering results.

Reply:
Oh...
I forgot to turn it off.
I was initially trying to manipulate the DOM using "aria-hidden" in JS, but didn't use it.

commit: [f8e317b](https://github.com/yasu-pro/js-handson/pull/18/commits/f8e317bacd47307ccbd17e4be7e3936fbf008a38)